### PR TITLE
feat(bucket): implement ACL rules

### PR DESCRIPTION
**Pull request by thesayyn**
Beware that I have disabled bucket before hook for a special reason. I will reenable them in a follow-up PR

### CODE REVIEW

File: stacks/api/bucket/expression/src/property_map.ts [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+  if (node.kind == "operator" && node.type == "select") {
+    const result = visitBinaryOperatorSelect(node);
+    return result ? [result] : [];
+  }
+  const lhs = visit(node.lhs);
+  const rhs = visit(node.rhs);
+  const result = [];
+
+  if (lhs) {
+    if (Array.isArray(lhs) && typeof lhs != "string") {
```
Comment (htuna07) [inline stacks/api/bucket/expression/src/property_map.ts:63]: The Second condition is unnecessary.

File: stacks/api/bucket/expression/src/property_map.ts [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+      result.push(...lhs);
+    } else {
+      result.push(lhs);
+    }
+  }
+
+  if (rhs) {
+    if (Array.isArray(rhs) && typeof rhs != "string") {
```
Comment (htuna07) [inline stacks/api/bucket/expression/src/property_map.ts:71]: The second condition is unnecessary.

File: stacks/api/bucket/expression/src/property_map.ts (hunk 1 above)
Comment (htuna07) [inline stacks/api/bucket/expression/src/property_map.ts:63]:
```suggestion
    if (Array.isArray(lhs)) {
```

File: stacks/api/bucket/expression/src/property_map.ts (hunk 2 above)
Comment (htuna07) [inline stacks/api/bucket/expression/src/property_map.ts:71]:
```suggestion
    if (Array.isArray(rhs)) {
```

File: stacks/api/bucket/src/bucket-data.controller.ts [hunk 3]
```diff
@@ -335,42 +315,60 @@ export class BucketDataController {
    * }
    * ```
    */
-  @UseInterceptors(activity(createBucketDataActivity))
+  //@UseInterceptors(activity(createBucketDataActivity))
```
Comment (htuna07) [inline stacks/api/bucket/src/bucket-data.controller.ts:318]:
```suggestion
  @UseInterceptors(activity(createBucketDataActivity))
```

File: stacks/spica/src/bucket/components/cel-editor/cel.syntax.ts [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+        /[a-z_$][\w$]*/,
+        {
+          cases: {
+            "@keywords": "keyword",
+            "@default": "identifier"
+          }
+        }
+      ],
+      [/[A-Z][\w\$]*/, "type.identifier"], // to show class names nicely
+      // [/[A-Z][\w\$]*/, 'identifier'],
```
Comment (htuna07) [inline stacks/spica/src/bucket/components/cel-editor/cel.syntax.ts:40]: We should remove or uncomment this line.

Comment (thesayyn) [reply stacks/spica/src/bucket/components/cel-editor/cel.syntax.ts:40]: i wanna keep this in case i need to reenable it for some reason

**Review by htuna07 — DISMISSED** [review]

**Pull request merged**